
project bootable/recovery/
diff --git a/bootable/recovery/crypto/fs_mgr/fs_mgr.c b/bootable/recovery/crypto/fs_mgr/fs_mgr.c
index 3aa9e60..fb54e38 100644
--- a/bootable/recovery/crypto/fs_mgr/fs_mgr.c
+++ b/bootable/recovery/crypto/fs_mgr/fs_mgr.c
@@ -649,6 +649,7 @@ out:
     return ret;
 }
 
+#define MAX_FAILED_COUNT 2
 /* If tmp_mount_point is non-null, mount the filesystem there.  This is for the
  * tmp mount we do to check the user password
  */
@@ -658,6 +659,7 @@ int fs_mgr_do_mount(struct fstab *fstab, char *n_name, char *n_blk_device,
     int i = 0;
     int ret = -1;
     char *m;
+    int failed_count = 0;
 
     if (!fstab) {
         return ret;
@@ -705,6 +707,9 @@ int fs_mgr_do_mount(struct fstab *fstab, char *n_name, char *n_blk_device,
                     fstab->recs[i].flags, fstab->recs[i].fs_options)) {
             ERROR("Cannot mount filesystem on %s at %s\n",
                     n_blk_device, m);
+            failed_count++;
+            if (failed_count < MAX_FAILED_COUNT)
+                continue;
             goto out;
         } else {
             ret = 0;
